certora/specs/SocialRecoveryModule.spec

```
@@ -1,26 +1,160 @@
 using SafeHarness as safeContract;
+using GuardianStorageHarness as guardianStorageContract;
+
+definition SENTINEL() returns address = 1;
 
 methods {
+    // Social Recovery Module Functions
+    function isGuardian(address, address) external returns (bool) envfree;
+    function guardiansCount(address) external returns (uint256) envfree;
+    function threshold(address) external returns (uint256) envfree;
+
+    // Guardian Storage Functions
+    function guardianStorageContract.countGuardians(address) external returns (uint256) envfree;
+
     // Safe Functions
-    function safeContract.isModuleEnabled(address) external returns (bool) envfree;
+    function safeContract.isModuleEnabled(address module) external returns (bool) envfree;
+    function safeContract.isOwner(address owner) external returns (bool) envfree;
+
+    // Wildcard Functions (Because of use of ISafe interface in Social Recovery Module)
+    function _.isModuleEnabled(address module) external => safeIsModuleEnabled(calledContract, module) expect bool ALL; // `calledContract` is a special variable.
+    function _.isOwner(address owner) external => sumarizeSafeIsOwner(calledContract, owner) expect bool ALL;
+}
+
+// A summary function to check if a module is enabled in the Safe contract.
```

I don't think this comment is accurate. I would argue that the summary doesn't check that the module is enabled. AFAIU, here we are using a summary to work around the fact that the recovery module uses the ISafe interface, and we want to assert that all the ISafe calls are done to the actual `safeContract`, and then it just returns what the `safeContract` would return.

```suggestion
// A summary function that asserts that all `ISafe.isModuleEnabled` calls are done
// to the `safeContract`, returning the same result as `safeContract`.
```

---

Updated 👍🏾